launchlib/launcher.go

```diff
@@ -87,30 +97,58 @@ func compileCmdFromConfig(
 	var executableErr error
 
 	if staticConfig.Type == "java" {
-		javaHome, javaHomeErr := getJavaHome(staticConfig.JavaConfig.JavaHome)
-		if javaHomeErr != nil {
-			return nil, javaHomeErr
-		}
-		_, _ = fmt.Fprintf(logger, "Using JAVA_HOME: %s\n", javaHome)
-
-		classpath := joinClasspathEntries(absolutizeClasspathEntries(workingDir,
-			staticConfig.JavaConfig.Classpath))
-		_, _ = fmt.Fprintf(logger, "Classpath: %s\n", classpath)
-
 		var combinedJvmOpts []string
 		combinedJvmOpts = append(combinedJvmOpts, staticConfig.JavaConfig.JvmOpts...)
 		combinedJvmOpts = append(combinedJvmOpts, customConfig.JvmOpts...)
 
-		jvmOpts := createJvmOpts(combinedJvmOpts, customConfig, logger)
-
-		executable, executableErr = verifyPathIsSafeForExec(path.Join(javaHome, "/bin/java"))
-		if executableErr != nil {
-			return nil, executableErr
+		// Handle experimental nativeImage execution mode before standard processing
+		if customConfig.Experimental.ExecutionMode == ExecutionModeNative {
+			executable, executableErr = verifyPathIsSafeForExec(customConfig.Experimental.NativeImageExecutablePath)
+			if executableErr != nil {
+				return nil, executableErr
+			}
+			args = append(args, executable) // 0th argument is the command itself
+
+			args = append(args, getNativeArgsFromJVMOpts(combinedJvmOpts)...)
+
+			args = append(args, customConfig.Experimental.NativeImageArguments...)
```

Comment:
From the existing code, I deduct that the JVM will use the last value for an option if it's present multiple times. Have we confirmed that the native image would behave in the same way? (might also be worth a quick comment explaining that we're porting over some specific jvm options from the defaults/custom, but also prioritizing any option that is explicitly set for native images)